src/main/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialImpl.java

```diff
@@ -138,6 +141,14 @@ private synchronized Token refreshToken(String oauthServerURL, String caCertData
         return extractTokenFromLocation(location.getValue());
     }
 
+    private String getOauthServerUrl(String apiServerURL, String caCertData, boolean skipTLSVerify) throws URISyntaxException, IOException, HttpClientWithTLSOptionsFactory.TLSConfigurationError {
+        URI uri = new URI(apiServerURL);
+        final HttpClientBuilder builder = HttpClientWithTLSOptionsFactory.getBuilder(uri, caCertData, skipTLSVerify);
+        HttpGet discover = new HttpGet(apiServerURL + "/.well-known/oauth-authorization-server");
```

Comment:
Do all API servers answer to such request in OpenShift ? Or do we need to fallback to the `apiServerURL + /oauth/authorize` when it's not the case ?

Reply:
Yes, I tested this api in OCP3 and OCP4 , it should be a standard for oauth authentication as reported [here](https://tools.ietf.org/html/draft-ietf-oauth-discovery-06#section-3.1)